### PR TITLE
Unify docs index pages and make sure they are always in sync with the menu

### PR DESCRIPTION
**Pull request by tnozicka**
**Description of your changes:**
This PR makes sure, that our toctree in the left menu matches the index pages and removes the inconsistencies, like when some links were named differently. This also eliminates the possibility that we miss some links that are present in the menu.

While it's not as pretty as when written manually, the consistency should prevail + I've filed https://github.com/scylladb/sphinx-scylladb-theme/issues/935

/cc @rzetelskik 
(on your request)

### CODE REVIEW

File: docs/source/multidc/index.rst [hunk 1]
```diff
@@ -5,21 +5,14 @@ Deploying multi-datacenter ScyllaDB clusters in Kubernetes
 Prepare a platform for a multi datacenter ScyllaDB cluster deployment:
 
 .. toctree::
-   :hidden:
```
Comment (rzetelskik) [inline docs/source/multidc/index.rst:8]: let's also lower maxdepth to 1 to align with what it looked like before

File: docs/source/multidc/index.rst [hunk 2]
```diff
@@ -5,21 +5,14 @@ Deploying multi-datacenter ScyllaDB clusters in Kubernetes
...
    eks
    gke
 
-* :doc:`Build multiple Amazon EKS clusters with Inter-Kubernetes networking <eks>`
-* :doc:`Build multiple GKE clusters with Inter-Kubernetes networking <gke>`
-
 Deploy a multi-datacenter ScyllaDB cluster in Kubernetes:
 
 .. toctree::
-   :hidden:
```
Comment (rzetelskik) [inline docs/source/multidc/index.rst:20]: let's also lower maxdepth to 1 to align with what it looked like before

File: docs/source/nodeoperations/index.rst [hunk 3]
```diff
@@ -3,20 +3,10 @@ Node operations using Scylla Operator
 ======================================
 
 .. toctree::
-   :hidden:
```
Comment (rzetelskik) [inline docs/source/nodeoperations/index.rst:6]: let's also lower maxdepth to 1 to align with what it looked like before

File: docs/source/multidc/index.rst (hunk 1 above)
Comment (tnozicka) [reply docs/source/multidc/index.rst:8]: done

File: docs/source/multidc/index.rst (hunk 2 above)
Comment (tnozicka) [reply docs/source/multidc/index.rst:20]: done

File: docs/source/nodeoperations/index.rst (hunk 3 above)
Comment (tnozicka) [reply docs/source/nodeoperations/index.rst:6]: done

**Comment by rzetelskik** [comment]
Thanks Tomas
\lgtm

**Comment by rzetelskik** [comment]
Thanks Tomas
/lgtm

**Pull request merged**